webapp/components/CommentList/CommentList.vue

```diff
@@ -40,5 +40,12 @@ export default {
   props: {
     post: { type: Object, default: () => {} },
   },
+  methods: {
+    deleteComment(deleted) {
+      this.post.comments = this.post.comments.map(comment => {
+        return comment.id === deleted.id ? deleted : comment
+      })
+    },
+  },
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T12:04:33Z" title="Tuesday, September 3rd 2019, 2:04:33 pm +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -40,5 +40,12 @@ export default {
   props: {
     post: { type: Object, default: () => {} },
   },
+  methods: {
+    deleteComment(deleted) {
+      this.post.comments = this.post.comments.map(comment => {
+        return comment.id === deleted.id ? deleted : comment
+      })
+    },
+  },
```


maybe this is what updates the list?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T12:56:50Z" title="Tuesday, September 3rd 2019, 2:56:50 pm +02:00">Sep 3, 2019</time>_

---


Exactly. A deleted comment does not get replaced (you would use `Array.filter`) but replaced with the response of `DeleteComment` (use `Array.map` and replace when found). The server response is called `deleted`, this is the argument of the `this.$emit` call.

If find this technique a much better separation of responsibilities. One component does the mutation and hands over the server response to the component which renders the list.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T14:02:06Z" title="Tuesday, September 3rd 2019, 4:02:06 pm +02:00">Sep 3, 2019</time>_

---


I like it as well, less code as well
Also, more consistency in the code base, how it is handled in one place, it is handled in another